mono/mini/mini-exceptions.c

```diff
@@ -1446,12 +1449,20 @@ mono_crash_reporting_register_native_library (const char *module_path, const cha
 	native_library_whitelist = g_list_append (native_library_whitelist, entry);
 }
 
+static void
+mono_crash_reporting_allow_all_native_libraries ()
+{
+	allow_all_native_libraries = TRUE;
+}
+
 static gboolean
 check_whitelisted_module (const char *in_name, const char **out_module)
 {
 #ifndef MONO_PRIVATE_CRASHES
 		return TRUE;
 #else
+	if (allow_all_native_libraries)
```

Comment:
This code has to set out_module, or the caller of this function is a no-op.

Reply:
@leculver could you please explain? From what I see, it is benign for callers to not set out_module, including when the frame information is written out. That said, we may just want to set it to `g_strdup (in_name)`.

Reply:
I may be misinterpreting this, I haven't had a chance to step through/debug it.  If out_module isn't set I think we do not take this branch, leaving `has_name` to false:

https://github.com/mono/mono/blob/cb83160ea4d60f030e71bcea60b19baa06f82192/mono/mini/mini-exceptions.c#L1749

If `has_name` isn't set, we end up bypassing writing anything out here:

https://github.com/mono/mono/blob/d54a6544b3165f13bb9cd9b5585648b0ac708622/mono/utils/mono-state.c#L532

Unless `str_descr` is originally set to a meaningful value before the function call to `check_whitelisted_module`?

In any case, testing should show whether things are being written out or not.

Reply:
This code is a bit confusing, so I'm not sure how it will behave, but I just want to mention @alexischr that you shouldn't call `g_strdup (in_name)` - all the crash reporting stuff is running in async (signal handler) context, so no allocating memory.